### PR TITLE
Revert "[Codegen] Add vector transfer + slice foldings in GenericVectorization (#17613)"

### PR DESCRIPTION
**Pull request by banach-space**
This reverts commit 8b8342596bb399f0c699e980d87b17875905d66e.

This change is hurting SVE+SME performance pretty badly. See
https://github.com/iree-org/iree/pull/17613 for context.

### CODE REVIEW

**Review by qedawkins — APPROVED** [review]
(Max is out this week so don't need to wait for his approval)

**Review by hanhanW — APPROVED** [review]

**Pull request merged**
